### PR TITLE
Encrypted backend v1.12.0

### PR DESCRIPTION
**Pull request by myyk**

### CODE REVIEW

File: router/rootfs/etc/confd/templates/backend.conf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+proxy_ssl_verify        off;
```
Comment (myyk) [inline router/rootfs/etc/confd/templates/backend.conf]: I could probably just get rid of `backend.conf` completely and move `proxy_ssl_verify        off;` into `nginx.conf`. Should I bother?

**Comment by myyk** [comment]
It's from this change that I built out Deis Router that is being deployed to Prod tonight. We can still clean it up more or not. I'm not sure. Some of the structure is there to make it easy to drop in more configurability that's not there now and not needed by us at this time.

**Comment by myyk** [comment]
cc @cyc @greglook @hobbeswalsh

File: router/rootfs/etc/confd/templates/backend.conf (hunk 1 above)
Comment (hobbeswalsh) [inline router/rootfs/etc/confd/templates/backend.conf]: I think this behavior belongs in `nginx.conf` -- without knowing too much about the way the `nginx.conf` file is currently structured, it's usually the place for proxy config, timeouts, etc.

File: router/rootfs/etc/confd/templates/nginx.conf [hunk 2]
```diff
@@ -282,7 +288,11 @@ http {
             set_if_empty $arg_{{ getv "/deis/router/affinityArg" }} $prng;
             {{ end }}
 
-            proxy_pass                  http://{{ $app }};
+            {{ if eq $proxySsl "true" }}
```
Comment (hobbeswalsh) [inline router/rootfs/etc/confd/templates/nginx.conf:292]: ah, this feels _so_ much better than the way the config is currently written. :+1:

File: router/rootfs/etc/confd/templates/backend.conf (hunk 1 above)
Comment (sbuss) [inline router/rootfs/etc/confd/templates/backend.conf]: :+1: to moving it to `nginx.conf`

**Comment by cyc** [comment]
Ping

**Comment by myyk** [comment]
@sbuss @hobbeswalsh @cyc Um... apparently I never made this official. I addressed all comments.

File: router/rootfs/etc/confd/templates/nginx.conf [hunk 3]
```diff
@@ -223,6 +224,12 @@ http {
         include deis.conf;
         {{ end }}
 
+        {{/* if a proxying SSL is setup for this app */}}
```
Comment (hobbeswalsh) [inline router/rootfs/etc/confd/templates/nginx.conf]: /s/a proxying/proxying

Also, I don't think this comment is as important as the one that's missing below.

File: router/rootfs/etc/confd/templates/nginx.conf [hunk 4]
```diff
@@ -223,6 +224,12 @@ http {
...
+        {{ if eq $proxySsl "true" }}
+        proxy_ssl_verify        off;
```
Comment (hobbeswalsh) [inline router/rootfs/etc/confd/templates/nginx.conf:229]: There should be a comment here about why verification is turned off.

File: router/rootfs/etc/confd/templates/nginx.conf [hunk 5]
```diff
@@ -300,6 +311,12 @@ http {
         server_name ~^{{ $app }}\.(?<domain>.+)$;
         include deis.conf;
 
+        {{/* if a proxying SSL is setup for this app */}}
+        {{ if eq $proxySsl "true" }}
+        proxy_ssl_verify        off;
```
Comment (hobbeswalsh) [inline router/rootfs/etc/confd/templates/nginx.conf:316]: explain why we're turning off verification

File: router/rootfs/etc/confd/templates/nginx.conf [hunk 6]
```diff
@@ -300,6 +311,12 @@ http {
         server_name ~^{{ $app }}\.(?<domain>.+)$;
         include deis.conf;
 
+        {{/* if a proxying SSL is setup for this app */}}
```
Comment (hobbeswalsh) [inline router/rootfs/etc/confd/templates/nginx.conf]: comment probably not needed

**Comment by hobbeswalsh** [comment]
Looks good modulo a few comments

**Comment by hobbeswalsh** [comment]
LGTM, thanks!

:shipit:

**Pull request merged**